### PR TITLE
Added android 13 in nightly build

### LINKED ISSUE #3139: Add Android 13  (API - 33) in the the nightly

### PR DESCRIPTION
**Pull request by MohitMaliFtechiz**
Fixes #3139

### CODE REVIEW

**Comment by kelson42** [comment]
I don't understand why the `nightly` workflow runs all the test again. The branch is already tested properly with the CI/Code coverage workflow?! To me this workflow should only build the nightly and push it `download.kiwix.org`. It could be IMO considered to somehow merge it with the release workflow. Can someone explain me why we run all the tests? On different APIs here?

**Comment by gouri-panda** [comment]
@kelson42 Different level API means different versions of Android. We are running all the UI tests in different versions of android to make sure everything work as expected in different types of versions of Android also it would be good idea to test the recently develop branch in nightly. We are only running unit tests once they are no need for different version of Android to test this.

File: .github/workflows/nightly.yml
Comment (gouri-panda) [inline .github/workflows/nightly.yml:13]: 32 is unnecessary.

File: .github/workflows/nightly.yml
Comment (gouri-panda) [inline .github/workflows/nightly.yml:26]: What's up with the default one?

File: .github/workflows/nightly.yml
Comment (MohitMaliFtechiz) [reply .github/workflows/nightly.yml]: hi @gouri-panda , done

File: .github/workflows/nightly.yml
Comment (gouri-panda) [reply .github/workflows/nightly.yml:26]: Ooh! It's required now.

**Review by gouri-panda — APPROVED** [review]

**Comment by kelson42** [comment]
> to make sure everything

How this run is different from what is run on `master` once a PR is merged?

**Comment by kelson42** [comment]
Without a clear reason why we should redundantly run this automated test in the `nightly`, I won't merge this PR. Instead I will request that the `nightly` workflow focus only on building and upload the API to download.kiwix.org.

**Comment by MohitMaliFtechiz** [comment]
> 

hi @kelson42 ,

On Recent time we had observed test are sometimes works on branch , while after merging it its started randomly failed on certain API level and we had placed lots of fix and enhance test cases according to that . So it will be good idea to add this variant to nightly.  

Yes nightly should focus on uploading artificat on ```download.kiwix.org``` but it were great help also to identify test which are often fail on nightly.

**Comment by kelson42** [comment]
> On Recent time we had observed test are sometimes works on branch , while after merging it its started randomly failed on certain API level and we had placed lots of fix and enhance test cases according to that . So it will be good idea to add this variant to nightly.

This is never a good idea to test twice the very same thing. How this test is different from what is secured with https://github.com/kiwix/kiwix-android/blob/develop/.github/workflows/coverage.yml#L7 ?
 
> Yes nightly should focus on uploading artificat on `download.kiwix.org` but it were great help also to identify test which are often fail on nightly.

Explain me how a test should fail differently on nightly considering you make the exact very same test on the exact very same code with https://github.com/kiwix/kiwix-android/blob/develop/.github/workflows/coverage.yml#L7 ? Which new info do you gather doing this?

**Comment by gouri-panda** [comment]
@kelson42 The main benefits form having the Nightly Builds: The code that has been committed during the day can be tested and reveal any recent problems (with the code, tests, or the infrastructure), and it is easy to identify the root cause of each problem to make sure new code doesn't break any old code. It is essential for the development to identify the issue early.

**Comment by kelson42** [comment]
@gouri-panda This is exactly what is done each time we merge to master. Please keep onky building and uploading in this workflow.

**Comment by gouri-panda** [comment]
>This is exactly what is done each time we merge to master.

@kelson42 Do you mean master or develop? Do you mean merging PR's into develop?

**Comment by kelson42** [comment]
Yes, i mean "develop"

**Comment by gouri-panda** [comment]
>Instead I will request that the nightly workflow focus only on building and upload the API to download.kiwix.org.

@kelson42 Ok, Sure!

**Pull request closed without merging**
